File: packages/ui/src/schemas/notifications.ts
import { z } from 'zod';

export const NotificationSchema = z.object({
  id: z.string(),
  type: z.string(),
  title: z.string(),
  message: z.string(),
  application_id: z.string().nullable(),
  is_read: z.boolean(),
  created_at: z.string(),
});

export const NotificationListSchema = z.object({
  items: z.array(NotificationSchema),
  total: z.number(),
  unread_count: z.number(),
});

export const UnreadCountSchema = z.object({
  unread_count: z.number(),
});

export type Notification = z.infer<typeof NotificationSchema>;
export type NotificationList = z.infer<typeof NotificationListSchema>;
export type UnreadCount = z.infer<typeof UnreadCountSchema>;

File: packages/api/src/schemas/notifications.py
"""
Notification Pydantic schemas.
"""

from datetime import datetime

from pydantic import BaseModel


class NotificationResponse(BaseModel):
    """Single notification response."""

    id: str
    type: str
    title: str
    message: str
    application_id: str | None = None
    is_read: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class NotificationListResponse(BaseModel):
    """Paginated notification list."""

    items: list[NotificationResponse]
    total: int
    unread_count: int


class UnreadCountResponse(BaseModel):
    """Unread notification count."""

    unread_count: int
